### ISSUE TITLE
VerifyRouting crashes if no route found

### ISSUE THREAD

**Issue by JayBazuzi** [approvals/ApprovalTests.Net#127]
1. In MvcApplication.Razor/Global.asax.cs, remove the default route
2. Run unit tests

Results:

NullReferenceException in AspApprovals.VerifyRouting(), on this line:

```
sb.AppendFormat("{0} => {1} \r\n", url, route.Values.ToReadableString());
```

My proposed fix is to null-coalesce the `RouteData`:

```
var route = routes.GetRouteData(httpContext) ?? new RouteData();
```

**Comment by isidore** [comment]
maybe.
I agree it shouldn't null pointer,
 but wouldn't this be more of a 404 error in production?  maybe the approval should display something more representing that?

**Comment by isidore** [comment]
possibly fixed with 58e3b895f4d33194481b02a6a7b4268f5aa33f90
